### PR TITLE
Add support for Tag, Semvar and Commit to data.flux_sync

### LINKED ISSUE #161: Add tag support in flux_sync
Hey,

Currently flux_sync accept only branch name and it is not possible to use a tag.

When using branch flux will fetch "refs/heads/<branch>"
When using tag flux has to fetch "refs/tags/<tag>"

As you can see those are not interchangeable.

Flux cli already has support for tag and tag-semver as described here: https://fluxcd.io/docs/cmd/flux_create_source_git/

Currently new inactive branch must be created to pin deployment to commit_id, using tags would resolve this issue for us.

### PR DESCRIPTION
**Pull request by philnichol**
Closes #161 

This allows users to specify tag, semver and commit via terraform

Duplicate #196 since I messed my git history up beyond repair

Signed-off-by: Phil Nichol <35630607+philnichol@users.noreply.github.com>

### CODE REVIEW

File: pkg/provider/data_sync.go [hunk 1]
```diff
@@ -64,6 +64,21 @@ func DataSync() *schema.Resource {
...
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"semver": {
+				Description: "The Git tag semver expression, takes precedence over `tag`.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"commit": {
+				Description: "The Git commit SHA to checkout, if specified Tag filters will be ignored.",
```
[the remark this answers is not in the document]
Comment (phillebaba) [reply pkg/provider/data_sync.go:78]: That is a valid point. Don't really see how we can solve this without removing the default branch value. One option could be to have the other values override the branch but I dont see how that can be enforced during the plan step.

[the remark this answers is not in the document]
Comment (stefanprodan) [reply pkg/provider/data_sync.go:78]: We've fixed this in flux 0.21, the branch is no longer required.

**Review by stefanprodan — APPROVED** [review]
LGTM

Thanks @philnichol 🥇

**Comment by philnichol** [comment]
Thanks @stefanprodan , have rebased and force pushed to a single commit :)

**Pull request merged**